compiler/ghc/default.nix

```diff
@@ -216,7 +246,10 @@ stdenv.mkDerivation (rec {
         # GHC is a bit confused on its cross terminology, as these would normally be
         # the *host* tools.
         export CC="${targetCC}/bin/${targetCC.targetPrefix}cc"
-        export CXX="${targetCC}/bin/${targetCC.targetPrefix}cxx"
+        ${
+          if builtins.compareVersions ghc-version "9.4" < 0
+            then ''export CXX="${targetCC}/bin/${targetCC.targetPrefix}cxx"''
+            else ''export CXX="${targetCC}/bin/${targetCC.targetPrefix}c++"''}
```

Comment:
We can't set c++ for GHC < 9.4?

Reply:
I think I did that just to avoid rebuilding the old GHCs while I was testing.  I'll include it unconditionally.